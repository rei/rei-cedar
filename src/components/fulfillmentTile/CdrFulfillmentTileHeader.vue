<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import CdrSubheadingSans from '../text/presets/CdrSubheadingSans.vue';
import CdrFulfillmentTileLayout from './CdrFulfillmentTileLayout.vue';
import { surfaceSelectionLayout } from '../../types/interfaces';

/** Header component with icon slot for fulfillment tile */

defineOptions({ name: 'CdrFulfillmentTileHeader' });

const style = useCssModule();
const baseClass = 'cdr-fulfillment-tile-header';

const rootProps = computed(
  (): surfaceSelectionLayout => ({
    orientation: 'horizontal',
    class: { [style[baseClass]]: true },
  }),
);
</script>

<template>
  <CdrFulfillmentTileLayout v-bind="rootProps">
    <span
      v-if="$slots['icon-left']"
      :class="style['cdr-fulfillment-tile-header__icon']"
    >
      <!-- @slot Place an icon to the left of the header content -->
      <slot name="icon-left" />
    </span>
    <CdrSubheadingSans
      :strong="true"
      scale="-1"
      tag="span"
      :class="style['cdr-fulfillment-tile-header__label']"
    >
      <!-- @slot Header content that is still visible during loading. -->
      <slot name="header" />
    </CdrSubheadingSans>
    <span
      v-if="$slots['icon-right']"
      :class="style['cdr-fulfillment-tile-header__icon']"
    >
      <!-- @slot Place an icon to the right of the header content -->
      <slot name="icon-right" />
    </span>
  </CdrFulfillmentTileLayout>
</template>

<style lang="scss" module src="./styles/CdrFulfillmentTileHeader.module.scss"></style>
