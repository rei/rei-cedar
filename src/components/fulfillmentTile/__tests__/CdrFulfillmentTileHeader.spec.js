import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrFulfillmentTileHeader from '../CdrFulfillmentTileHeader.vue';

describe('CdrFulfillmentTileHeader', () => {
  it('icon left configuration', () => {
    const wrapper = mount(CdrFulfillmentTileHeader, {
      slots: {
        'icon-left': 'X',
        header: 'Label slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('icon right configuration', () => {
    const wrapper = mount(CdrFulfillmentTileHeader, {
      slots: {
        'icon-right': 'X',
        header: 'Label slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('shows label', () => {
    const wrapper = mount(CdrFulfillmentTileHeader, {
      slots: {
        header: 'Label slot',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find('.cdr-fulfillment-tile-header__label').text()).toBe('Label slot');
  });
});
