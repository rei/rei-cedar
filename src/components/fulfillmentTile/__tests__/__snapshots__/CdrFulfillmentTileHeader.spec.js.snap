// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrFulfillmentTileHeader > icon left configuration 1`] = `
<div
  class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
>
  <!-- @slot Place elements in the default slot. -->
  
  <span
    class="cdr-fulfillment-tile-header__icon"
  >
    <!-- @slot Place an icon to the left of the header content -->
    
    X
    
  </span>
  <span
    class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
    strong="true"
    style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
  >
    
    <!-- @slot Header content that is still visible during loading. -->
    
    Label slot
    
    
  </span>
  <!--v-if-->
  
</div>
`;

exports[`CdrFulfillmentTileHeader > icon right configuration 1`] = `
<div
  class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
>
  <!-- @slot Place elements in the default slot. -->
  
  <!--v-if-->
  <span
    class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
    strong="true"
    style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
  >
    
    <!-- @slot Header content that is still visible during loading. -->
    
    Label slot
    
    
  </span>
  <span
    class="cdr-fulfillment-tile-header__icon"
  >
    <!-- @slot Place an icon to the right of the header content -->
    
    X
    
  </span>
  
</div>
`;

exports[`CdrFulfillmentTileHeader > shows label 1`] = `
<div
  class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal cdr-fulfillment-tile-header"
>
  <!-- @slot Place elements in the default slot. -->
  
  <!--v-if-->
  <span
    class="cdr-subheading-sans cdr-fulfillment-tile-header__label"
    strong="true"
    style="--cdr-subheading-sans-font-size: var(--cdr-type-scale--1); --cdr-subheading-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-subheading-sans-line-height-ratio));"
  >
    
    <!-- @slot Header content that is still visible during loading. -->
    
    Label slot
    
    
  </span>
  <!--v-if-->
  
</div>
`;
